### ISSUE TITLE
Grid column headers are not aligned with the columns themselves

### ISSUE THREAD

**Issue by archiecobbs** [vaadin/flow-components#6849]
### Description

After loading and displaying a new `Grid`, the column headers are not aligned with the columns themselves.

If you then grab one of the separators between the column headers and slide it left or right, the headers "snap" into their correct alignments.

Here's a video showing the problem:

https://github.com/user-attachments/assets/230c1121-af7c-47c8-a382-9d8dd13e0e9e

### Expected outcome

The column headers should always be aligned with the columns themselves.

### Minimal reproducible example

Please download and run this reproducer: https://github.com/archiecobbs/vaadin-grid-load-fail-issue/tree/column-alignment-issue

Note you have to checkout the **column-alignment-issue** branch.

### Steps to reproduce

1. Run `git clone git@github.com:archiecobbs/vaadin-grid-load-fail-issue.git`
1. Run `git checkout column-alignment-issue`
1. You need to have `tomcat@10` installed via `brew`
1. Run `mvn clean package`
1. Run the script `./run.sh`
1. Connect to http://localhost:8080/example/example/

### Environment

Vaadin version(s): 24.5.5
OS: MacOS 15.1.1

### Browsers

Chrome

**Comment by archiecobbs** [comment]
> Hi @archiecobbs, the issue you're seeing is caused by the use of em units in column widths.

Thanks for the hint @rolfsmeds !

But bleh... this is yet another obscure Vaadin booby trap. I've fallen into so many over the years...

So I'll change this bug to be a feature request - to log a warning in this situation.

Thanks.

**Comment by rolfsmeds** [comment]
Yes, it's a very sneaky booby trap that you'll only be aware of in case you happen to read the [javadoc](https://github.com/vaadin/flow-components/blob/36842be4121732958ffaa2860f6ff105ed0bcfb6/vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java#L582-L585) or the [Grid docs page on column width](https://vaadin.com/docs/latest/components/grid/columns#column-width) 😉 

I'll recategorize this as a feature request then.

Oh, and suggestions of how to remove said booby trap are always welcome. Preferably ones that avoid introducing breaking changes affecting existing applications.
